lib/rethinkdb_ecto/repo.ex

```diff
@@ -72,6 +74,30 @@ defmodule RethinkDB.Ecto.Repo do
     end
   end
 
+  def convert_date(item) do
+    {attr, value} = item
+    date_string = to_date_string(value)
+
+    cond do
+      date_string -> {attr, to_rethinkdb_date(date_string)}
+      true -> item
+    end
+  end
+
+  defp to_rethinkdb_date(value), do: RethinkDB.Query.iso8601(value)
+
+  defp to_date_string(%Ecto.Date{year: year, month: month, day: day}) do
+    %Ecto.Date{year: year, month: month, day: day}
```

Comment:
If we're breaking this out already, I would look at the `time` query in RethinkDB. There's a 4 arg version and a 7 args version.